pkg/operator/internal/webhooks/kubecarrier_webhook.go

```diff
@@ -70,5 +71,33 @@ func (r *KubeCarrierWebhookHandler) validateCreate(kubeCarrier *operatorv1alpha1
 	if kubeCarrier.Name != constants.KubeCarrierDefaultName {
 		return fmt.Errorf("KubeCarrier object name should be 'kubecarrier', found: %s", kubeCarrier.Name)
 	}
+	auth := map[string]bool{}
+	for _, a := range kubeCarrier.Spec.API.Authentication {
+		if a.OIDC != nil {
```

Comment:
also check that one and only one field is set up (e.g. have a counter how many fields are setup, increase by one, and at the end verify that the counter is equal to 1).

Reply:
done